src/phoenix/trace/trace_dataset.py

```diff
@@ -133,3 +150,35 @@ def to_disc(self) -> None:
             allow_truncated_timestamps=True,
             coerce_timestamps="ms",
         )
+
+    def append_evaluations(self, evaluations: TraceEvaluations) -> None:
+        """adds an evaluation to the traces"""
+        # Append the evaluations to the list of evaluations
+        self.evaluations.append(evaluations)
+
+    def get_evals_dataframe(self) -> DataFrame:
+        """
+        Creates a flat dataframe of all the evaluations for the dataset.
+        """
+        return pd.concat(
+            [evals.get_dataframe(prefix_columns_with_name=True) for evals in self.evaluations],
+            axis=1,
+        )
+
+    def get_spans_dataframe(self, include_evaluations: bool = True) -> DataFrame:
+        """
+        converts the dataset to a dataframe of spans. If evaluations are included,
+        the evaluations are merged into the dataframe.
+
+        Parameters
+        __________
+        include_evaluations: bool
+            if True, the evaluations are merged into the dataframe
+        """
+        df = self.dataframe.copy()
+        if not include_evaluations:
+            return df
+        evals_df = self.get_evals_dataframe()
+        # Make sure the index is set to the span_id
+        df.set_index(EVALUATIONS_INDEX_NAME, drop=False, inplace=True)
```

Comment:
I'm gonna do it here on the copy so that I'm not messing with the main dataframe

Reply:
It should be fine because by design almost all operation in pandas return a copy. You can see d1 and d2 are different objects below.

<img width="282" alt="Screenshot 2023-12-01 at 12 17 01 PM" src="https://github.com/Arize-ai/phoenix/assets/80478925/1f7138f8-66d0-459b-9f8c-ace67c8d0125">